File: 060/main.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function (n, k) {
  let list = []
  let fac = [1]
  let res = []
  for (let i = 0; i < n; i++) {
    list.push(i + 1)
  }
  k--
  for (let i = 1; i < n; i++) {
    let tmp = fac[i - 1] * i
    fac.push(tmp)
  }
  for (let i = n - 1; i >= 0; i--) {
    let num = Math.floor(k / fac[i])
    res.push(list[num])
    list.splice(num, 1)
    k = k % fac[i]
  }
  return res.join('')
}

console.log(getPermutation(4, 4))

File: 060/main.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    string getPermutation(int n, int k) {
        vector<int> v;
        int fac[n];
        string res = "";
        for (int i = 0; i < n; i++) {
            v.push_back(i + 1);
        }
        fac[0] = 1;
        for (int i = 1; i < n; i++) {
            fac[i] = fac[i - 1] * i;
        }
        k--;
        for (int i = n - 1; i >= 0; i--) {
            int num = k / fac[i];
            res += v[num] + '0';
            v.erase(v.begin() + num);
            k = k % fac[i];
        }
        return res;
    }
};

int main()
{
    Solution sol;
    string res = sol.getPermutation(4, 4);
    cout << res << endl;
    return 0;
}
